hedera-node/hedera-mono-service/src/main/java/com/hedera/services/contracts/execution/EvmTxProcessor.java

```diff
@@ -124,16 +124,32 @@ protected TransactionProcessingResult execute(
                 new HederaTracer(!isStatic && isSideCarTypeEnabled(SidecarType.CONTRACT_ACTION));
 
         super.setOperationTracer(hederaTracer);
-        super.execute(
-                sender,
-                receiver,
-                gasPrice,
-                gasLimit,
-                value,
-                payload,
-                contractCreation,
-                isStatic,
-                mirrorReceiver);
+
+        try {
+            super.execute(
+                    sender,
+                    receiver,
+                    gasPrice,
+                    gasLimit,
+                    value,
+                    payload,
+                    contractCreation,
+                    isStatic,
+                    mirrorReceiver);
+        } catch (ResourceLimitException e) {
+            handleResourceLimitExceeded(
```

Comment:
after handling a 'resource limit exceeded' you're always going to throw a proper exception for it - so make it one step?  (`handleResourceLimitExceeded` to return the exception to throw?)

Reply:
By 'throwing a proper exception` you have in mind returning the `TransactionProcessingResult`, i.e. combining the 2 functions, right? I wanted to somewhat separate the actions of 'teardown' after processing and the return of the specific processing result, but the suggested approach (if I'm understanding correctly) sounds good to me also